src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -82,6 +140,69 @@ export function getEstimates(patientSurvivals: PatientSurvival[]): number[] {
     return estimates;
 }
 
+export function parseSurvivalData(s: string) {
+    const splitStatusAndLabel = s.split(':').map(text => text.trim());
+    let status = undefined;
+    let label = undefined;
+    // survival data in new format "status:label"
+    if (splitStatusAndLabel.length === 2) {
+        status = splitStatusAndLabel[0];
+        label = splitStatusAndLabel[1];
+    } else {
+        // old format "label"
+        label = splitStatusAndLabel[0];
+    }
+
+    // if status not exists, status will be undefined
+    return {
+        status,
+        label,
+    } as ParsedSurvivalData;
+}
+
+export function getSurvivalStatusBoolean(s: string, prefix: string) {
+    // if prefix is in predefined survival type
+    if (prefix in survivalClinicalDataVocabulary) {
+        return survivalClinicalDataVocabulary[prefix].includes(s);
+    }
+    // if prefix is custom survival type
+    else {
+        const parsedSurvivalData = parseSurvivalData(s);
+        if (parsedSurvivalData.status) {
+            return parsedSurvivalData.status === '1';
+        }
+    }
+    // return false as default for values that cannot get mapped
+    return false;
+}
+
+export function getStatusCasesHeaderVocabulary(
```

Comment:
i don't think "vocabulary" is the right word here.  it's just a string so it should be "text"